modules/data.land/R/clip_and_move_raster_file.R

```diff
@@ -0,0 +1,54 @@
+#' Clip and Move a Raster File
+#'
+#' Clips a raster to a polygon bounding box, optionally masks to polygon, and saves the 
+#' output in the same format as the input.
+#'
+#' @param input_path Character. Path to the input raster file.
+#' @param polygon An object or file coercible to a `SpatVector` by `terra::vect()`
+#'   (e.g., an `sf` object, a `SpatVector`, or a file path to a vector dataset).
+#'   used for clipping and masking. Must have a valid CRS.
+#' @param out_path Character. Path to save the processed raster.
+#' @return Invisibly, the clipped `SpatRaster` object. The raster is also saved to `out_path`.
+#' @export
+#' @author David LeBauer
+clip_and_move_raster_file <- function(input_path, polygon, out_path, mask = TRUE, overwrite = TRUE) {
```

Comment:
Nit: What does the "and move" signify? I'm realizing each time I come back to this PR with fresh eyes I spend a few minutes wondering if this somehow alters the input file as well as creating a new clipped output. 

Would just `clip_raster_file` be clearer?